From 07fb8d2aeeeceadf6e340622b912675d852836a1 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Tue, 22 Jun 2021 20:31:46 -0500
Subject: [PATCH 395/456] power: supply: ip5xxx: Long press to turn off

The long-press action actually works on my hardware, unlike double-press.

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 drivers/power/supply/ip5xxx_power.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/power/supply/ip5xxx_power.c b/drivers/power/supply/ip5xxx_power.c
index 7f727f5be80b..ae5d3167594e 100644
--- a/drivers/power/supply/ip5xxx_power.c
+++ b/drivers/power/supply/ip5xxx_power.c
@@ -525,12 +525,14 @@ static void ip5xxx_power_wd_work(struct work_struct *work)
 
 	/*
 	 * Enable the NTC.
-	 * Configure the button for long press => LED, two press => shutdown.
+	 * Configure the button for two press => LED, long press => shutdown.
 	 */
 	ret = regmap_update_bits(ip5xxx->regmap, IP5XXX_SYS_CTL5,
 				 IP5XXX_SYS_CTL5_NTC_EN |
 				 IP5XXX_SYS_CTL5_WLED_MODE_SEL |
-				 IP5XXX_SYS_CTL5_BTN_SHDN_SEL, 0);
+				 IP5XXX_SYS_CTL5_BTN_SHDN_SEL,
+				 IP5XXX_SYS_CTL5_WLED_MODE_SEL |
+				 IP5XXX_SYS_CTL5_BTN_SHDN_SEL);
 	if (ret)
 		goto err;
 
-- 
2.34.1

